fahrschein/src/main/java/org/zalando/fahrschein/NakadiClient.java

```diff
@@ -54,6 +58,35 @@ public static NakadiClientBuilder builder(URI baseUri) {
         }
     }
 
+    public <T> void publish(String eventName, List<T> events) throws IOException {
+        final URI uri = baseUri.resolve(String.format("/event-types/%s/events", eventName));
+        final ClientHttpRequest request = clientHttpRequestFactory.createRequest(uri, HttpMethod.POST);
+
+        request.getHeaders().setContentType(MediaType.APPLICATION_JSON);
+
+        try (final OutputStream body = request.getBody()) {
+            objectMapper.writeValue(body, events);
+        }
+
+        try (final ClientHttpResponse response = request.execute()) {
+            if ((response.getRawStatusCode() == 207 || response.getRawStatusCode() == 422)) {
```

Comment:
I'd have to check if nakadi returns a specific content type for the `BatchItemResponse`, then we could skip the status code check. Other error status codes should already have been handled by `ProblemHandlingClientHttpRequestFactory`.